pandas/core/categorical.py

```diff
@@ -879,8 +874,26 @@ def rename_categories(self, new_categories, inplace=False):
 
         Parameters
         ----------
-        new_categories : Index-like or dict-like (>=0.21.0)
-           The renamed categories.
+        new_categories : list-like or dict-like
+           The categories end up with
+
+           .. versionchanged:: 0.21.0
+
+              new_categories may now also be dict-like, in which case it
+              specifies a mapping from old-categories to new.
+
+            If it is list-like, all items must be unique and the number of
+            items in the new categories must match the existing number of
+            categories.
+
+            If dict-like, categories not contained in the mapping are passed
+            through.
```

Comment:
I find this a bit unstructured with the versionchanged and the further explanation splitted.

Suggestion (just copy paste, not edited sentences):

```
- list-like: If it is list-like, all items must be unique and the number of
  items in the new categories must match the existing number of
  categories.
- .. versionadded:: 0.21.1 dict-like, in which case it specifies a mapping from old-categories to new. Categories not contained in the mapping are passed through.

.. warning::

   about series
```

(only not fully sure how the versionchanges works in a list)